### PR TITLE
Install with no arguments

### PR DESCRIPTION
**Pull request by gigabo**
Use temporary package.json files.

This is groundwork for:

- Pluggable npm clients (yarn).
- A `lerna upgrade` command.

This is a port of https://github.com/asini/asini/pull/45.  Please see original PR commentary during review.

### CODE REVIEW

File: src/NpmUtilities.js [hunk 1]
```diff
@@ -11,18 +13,62 @@ export default class NpmUtilities {
...
+        }
+      });
+    });
+  }
+
+  // Take a dep like "foo@^1.0.0".
+  // Return a tuple like ["foo", "^1.0.0"].
+  // Handles scoped packages.
+  // Returns undefined for version if none specified.
+  static splitVersion(dep) {
```
Comment (doug-wade) [inline src/NpmUtilities.js]: I share @rygine's [enthusiasm for regexes](https://github.com/asini/asini/pull/45/files#r85001506), but it seems that you could write tests for splitVersion directly that verified that it works as intended, which would make me feel much better.

File: src/FileSystemUtilities.js [hunk 2]
```diff
@@ -39,6 +39,16 @@ export default class FileSystemUtilities {
     fs.writeFile(filePath, ensureEndsWithNewLine(fileContents), callback);
   }
 
+  @logger.logifyAsync()
```
Comment (doug-wade) [inline src/FileSystemUtilities.js]: It seems like most [FileSystemUtilities have tests](https://github.com/lerna/lerna/blob/master/test/FileSystemUtilities.js); seems worthwhile to add some for these as well.

File: src/NpmUtilities.js (hunk 1 above)
Comment (gigabo) [inline src/NpmUtilities.js]: Good call.  Added [some tests](https://github.com/lerna/lerna/pull/576/files#diff-b4cd8e152e8c3ce22a41f4bae7d04939).

File: src/FileSystemUtilities.js (hunk 2 above)
Comment (gigabo) [inline src/FileSystemUtilities.js]: Another good call.  Added [some more tests](https://github.com/lerna/lerna/pull/576/files#diff-c2bcab92ef3ab638c3bb52c8a6cdf9e9).

**Pull request merged**